src/backend/distributed/planner/intermediate_result_pruning.c

```diff
@@ -154,7 +154,7 @@ RecordSubplanExecutionsOnNodes(HTAB *intermediateResultsHash,
 	List *usedSubPlanNodeList = distributedPlan->usedSubPlanNodeList;
 	List *subPlanList = distributedPlan->subPlanList;
 	ListCell *subPlanCell = NULL;
-	int workerNodeCount = ActiveReadableNonCoordinatorNodeCount();
+	int workerNodeCount = list_length(ActiveReadableNodeList());
```

Comment:
Just noting an idea that I considered earlier, but not suggesting as that is wrong.

If the coordinator is added but shouldhaveshards=false, we might unnecessarily send the intermediate results to the coordinator. However, that is not true because we only send to nodes that there are placements to join with the intermediate result (except for SUBPLAN_ACCESS_ANYWHERE, which is OK I guess)